src/bun.js/bindings/sqlite/JSSQLStatement.cpp

```diff
@@ -671,63 +806,174 @@ static inline bool rebindValue(JSC::JSGlobalObject* lexicalGlobalObject, sqlite3
+    const bool throwOnMissing = trimLeadingPrefix;
+
+    // Did they reorder the columns?
+    //
+    // { ?2: "foo", ?1: "bar" }
+    //
+    if (UNLIKELY(bindings.hasOutOfOrderNames)) {
+
+        const auto& getValue = [&](const char* name, size_t i) -> JSValue {
+            JSValue value = {};
+            if (name == nullptr) {
+                return target->getDirectIndex(globalObject, i);
+            }
 
-        JSValue value;
-        if (LIKELY(!slot.isTaintedByOpaqueObject()))
-            value = slot.getValue(globalObject, propertyName);
-        else {
-            value = target->get(globalObject, propertyName);
+            if (trimLeadingPrefix) {
+                name += 1;
+            }
+
+            const WTF::String str = WTF::String::fromUTF8ReplacingInvalidSequences({ reinterpret_cast<const unsigned char*>(name), strlen(name) });
+
+            if (trimLeadingPrefix && name[0] >= '0' && name[0] <= '9') {
+                auto integer = WTF::parseInteger<int32_t>(str, 10);
+                if (integer.has_value()) {
+                    return target->getDirectIndex(globalObject, integer.value() - 1);
+                }
+            }
+
+            const auto identifier = Identifier::fromString(vm, str);
+            PropertySlot slot(target, PropertySlot::InternalMethodType::GetOwnProperty);
+            if (!target->getOwnNonIndexPropertySlot(vm, &structure, identifier, slot)) {
+                return JSValue();
+            }
+
+            if (LIKELY(!slot.isTaintedByOpaqueObject())) {
+                return slot.getValue(globalObject, identifier);
+            }
+
+            return target->get(globalObject, identifier);
+        };
+
+        for (size_t i = 0; i < size; i++) {
+            auto* name = sqlite3_bind_parameter_name(stmt, i + 1);
+
+            JSValue value = getValue(name, i);
+            if (!value && !scope.exception()) {
+                if (throwOnMissing) {
+                    throwException(globalObject, scope, createError(globalObject, makeString("Missing parameter \""_s, name, "\""_s)));
+                } else {
+                    continue;
+                }
+            }
             RETURN_IF_EXCEPTION(scope, JSValue());
+
+            if (!rebindValue(globalObject, db, stmt, i + 1, value, scope, clone, safeIntegers)) {
+                return JSValue();
+            }
+
+            RETURN_IF_EXCEPTION(scope, {});
+            count++;
+        }
+    }
+    // Does it only contain indexed properties?
+    //
+    // { 0: "foo", 1: "bar", "2": "baz" }
+    //
+    else if (UNLIKELY(bindings.isOnlyIndexed)) {
+        for (size_t i = 0; i < size; i++) {
+            JSValue value = target->getDirectIndex(globalObject, i);
+            if (!value && !scope.exception()) {
+                if (throwOnMissing) {
+                    throwException(globalObject, scope, createError(globalObject, makeString("Missing parameter \""_s, i + 1, "\""_s)));
```

Comment:
If we throw or have an exception should we return? should we add  `RETURN_IF_EXCEPTION(scope, JSValue());` after the loop (in case we throw and skip all other bindings)

Reply:
Yes, it does that a few lines below

https://github.com/oven-sh/bun/blob/0ba290c5395b6d9393a1069b68d4d1aae64ae259/src/bun.js/bindings/sqlite/JSSQLStatement.cpp#L896

Reply:
But this is inside the loop right?

Reply:
https://github.com/oven-sh/bun/blob/c0ab3ef88b73d7e6805bc22ee98d8159e38d972e/src/bun.js/bindings/sqlite/JSSQLStatement.cpp#L977